### PR TITLE
no-identical-functions/S4144: ignore numerical and string literal

### LINKED ISSUE #599: Improve S4144: ignore numerical and string literals
[RSPEC-4144](https://jira.sonarsource.com/browse/RSPEC-4144) has been updated to ignore numerical and string literals. Thus such case should raise an issue:
```ts
class A {
    fnOne() {
        this.foo = this.given + 1;
        this.fnFoo("xXx", this.foo);
        this.fooArray.push(1);
        this.isFoo = true;
        this.fnFoo("yYy", this.foo);
    }

    fnTwo() {
        this.foo = this.given + 2;
        this.fnFoo("xXx", this.foo);
        this.fooArray.push(2);
        this.isFoo = true;
        this.fnFoo("yYy", this.foo);
    }
}
```

Fixes #596

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #599

### CODE REVIEW

**Comment by vilchik-elena** [comment]
@saberduck I'm thinking about removing rule from Sonar way, wdyt?

File: sonarts-core/src/utils/areEquivalent.ts [hunk 1]
```diff
@@ -32,7 +33,17 @@ const COMPARED_BY_TEXT = new Set([
...
 ]);
 
-export default function areEquivalent(first: ts.Node | ts.Node[], second: ts.Node | ts.Node[]): boolean {
+/**
+ *
+ * @param first
+ * @param second
+ * @param ignoreLiterals when true, two nodes are considered equivalent even if string or numeric literals have different value
+ */
+export default function areEquivalent(
```
Comment (SonarTech) [inline sonarts-core/src/utils/areEquivalent.ts]: Code Smell Code Smell: Refactor this function to reduce its Cognitive Complexity from 17 to the 15 allowed. ([typescript:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=625&issues=AWM_0mTiS2S-MqecJ7NA&open=AWM_0mTiS2S-MqecJ7NA)

[ISSUE_KEY]: AWM_0mTiS2S-MqecJ7NA

Comment (SonarTech) [inline sonarts-core/src/utils/areEquivalent.ts]: Code Smell Code Smell: Refactor this function to reduce its Cognitive Complexity from 17 to the 15 allowed. ([typescript:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=625&issues=AWM_0mTiS2S-MqecJ7NA&open=AWM_0mTiS2S-MqecJ7NA)

[ISSUE_KEY]: AWM_0mTiS2S-MqecJ7NA

**Comment by vilchik-elena** [comment]
@saberduck I've changed message and put rule back to Sonar way. (I had a look at issues on peach, and looks like there are many new issues on vscode only, on others not so many)

**Pull request merged**